pydarn/plotting/rtp.py

```diff
@@ -0,0 +1,438 @@
+    def __filter_data_check(cls, dmap_record: dict, settings: dict, j: int) -> bool:
+        """
+        checks for data that does not meet the criteria of the filtered
+        settings
+
+        Parameters
+        ----------
+        dmap_record : dict
+            dictionary of the dmap record fields
+        settings : dict
+            dictionary of the settings list
+        j : int
+            index on the slist or range gate to check the array values
+
+        Returns
+        -------
+        pass_flg : bool
+            boolean indicating if the dmap_record passes all filter checks
+        """
+        pass_flg = True
+        for key, value in settings['min_array_filter'].items():
+            if dmap_record[key][j] < value:
+                pass_flg = False
+                break
+        for key, value in settings['max_array_filter'].items():
+            if dmap_record[key][j] > value:
+                pass_flg = False
+                break
+
+        for key, value in settings['min_scalar_filter'].items():
+            if dmap_record[key] < value:
+                pass_flg = False
+                break
+
+        for key, value in settings['max_scalar_filter'].items():
+            if dmap_record[key] > value:
+                pass_flg = False
+                break
+
+        for key, value in settings['equal_scalar_filter'].items():
+            if dmap_record[key] != value:
+                pass_flg = False
+                break
+
+        return pass_flg
+
+    @classmethod
+    def __check_data_type(cls, parameter: str, expected_type: str):
+        """
+        Checks to make sure the plot type is correct
+        for the data structure
+
+        Parameters
+        ----------
+        paraneter: str
+            string key word name of the parameter
+        expected_type: str
+            string decsribing an array or scalar type
+            to determine which one is needed for the type of plot
+
+        Raises
+        -------
+        RTPIncorrectPlotMethodError
+        """
+        data_type = cls.dmap_data[0][parameter]
+        if expected_type == 'array':
+            if not isinstance(data_type, np.ndarray):
+                raise rtp_exceptions.RTPIncorrectPlotMethodError(parameter,
+                                                                 data_type)
+        else:
+            if isinstance(data_type, np.ndarray):
+                raise rtp_exceptions.RTPIncorrectPlotMethodError(parameter,
+                                                                 data_type)
+
+    # TODO: move to a utils or superDARN utils
+    @classmethod
+    def __time2datetime(cls, dmap_record: dict) -> datetime:
+        """
+        Converts DMAP time parameter fields into a datetime object
+
+        Parameter
+        ---------
+        dmap_record: dict
+            dictionary of the DMAP data contains the time data
+
+        Returns
+        -------
+        datetime:
```

Comment:
Since you are returning a function call perhaps say `datetime object` and omit the colon?

Reply:
done